### ISSUE TITLE
ListContainersSegmentedAsync Returns Zero Results For Specific Storage Account

### ISSUE THREAD

**Issue by mayoatbm** [Azure/azure-storage-net#818]
### Which service(blob, file, queue, table) does this issue concern?
Blobs

### Which version of the SDK was used?
9.3.1 & 9.3.2

### Which platform are you using? (ex: .NET Core 2.1)
.Net Core 2.1

### What problem was encountered?
We cannot list containers for a specific storage account using the code below. This same code successfully works against multiple other storage accounts and it fact it worked against the problematic one a few days ago as well. 

`var storageAccount = CloudStorageAccount.Parse(connectionString);
            var blobClient = storageAccount.CreateCloudBlobClient();

            var results = new List<string>();

            BlobContinuationToken continuationToken = null;

            do
            {
                var containers = blobClient.ListContainersSegmentedAsync(continuationToken).GetAwaiter().GetResult();

                results.AddRange(containers.Results.Select(x => x.Name));

            } while (continuationToken != null);

            Console.WriteLine(results.Count);`

I have confirmed that the connection string is correct multiple times. The only other extenuating factor i can think of is that we are currently running load tests that involve this storage account heavily and this problem only showed up after a load test run. We can list containers in the storage explorer and on Azure but for this one storage account, all our calls from the sdk return zero results. 

I don't want to put the storage account name here, but i'm happy to send some other piece of information that can help locate it if someone tells me what that is.

Listing containers in the storage account is the first step of an ingestion process we are load testing and it naturally doesn't run because it thinks there are no containers. Meanwhile the container has TBs of data in it.

### How can we reproduce the problem in the simplest way?
I need to communicate the name of the storage account to you somehow and you can try running the code above against it.

### Have you found a mitigation/solution?
We are currently blocked by this issue and are facing a very tight deadline. Please help.

**Comment by pemari-msft** [comment]
Can you please email the account and container name to us at ‘ascl [at] Microsoft [dot] com’?

**Comment by mayoatbm** [comment]
@pemari-msft  Done. Thanks for responding so quickly.

**Comment by pemari-msft** [comment]
From the code snippet, it looks like you’re never updating the value of the continuation token, and therefore not following it when it is returned by the service. Is that the case? Can you try updating the token or logging the value to check?

**Comment by mayoatbm** [comment]
For those coming to this later. There are two things going on here:

1. There's a bug in the code above (not updating the continuation token) that should have resulted in the  code only processing one page's worth of containers, but that’s not what happened at all because of the second issue.
2, The ListxxxSegmented method sometimes return zero results on the first segment (but with a non null continuation token). If you're expecting only less than a page's worth of data (e.g. get the first container) and don't follow the continuation token, you will get burned. 

In summary: ALWAYS follow the continuation token, no matter how many results you expect.

Thanks to @pemari-msft  for a very fast and effective response.

Closing...
